spec/system-probe_spec.rb

```diff
@@ -240,4 +240,115 @@
       })
     end
   end
+
+  context 'with CWS enabled' do
+    cached(:solo) do
+      ChefSpec::SoloRunner.new(
+        platform: 'ubuntu',
+        version: '16.04'
+      ) do |node|
+        node.name 'chef-nodename' # expected to be used as the hostname in `datadog.yaml`
+        node.normal['datadog'] = {
+          'api_key' => 'somethingnotnil',
+          'agent_major_version' => 6,
+          'security_agent' => {
+            'cws' => {
+              'enabled' => true,
+            }
+          },
+          'extra_config' => {
+            'security_agent' => {
+              'runtime_security_config' => {
+                'activity_dump' => {
+                  'enabled' => true,
+                }
+              }
+            }
+          }
+        }
+      end
+    end
+
+    cached(:chef_run) do
+      solo.converge(described_recipe) do
+        solo.resource_collection.insert(
+          Chef::Resource::Service.new('datadog-agent', solo.run_context))
+        solo.resource_collection.insert(
+          Chef::Resource::Service.new('datadog-agent-security', solo.run_context))
+      end
+    end
+
+    it 'system-probe.yaml is created' do
+      expect(chef_run).to create_template('/etc/datadog-agent/system-probe.yaml')
+    end
+
+    it 'system-probe.yaml contains expected YAML configuration' do
+      expected_yaml = <<-EOF
+      runtime_security_config:
+        enabled: true
+        activity_dump:
+          enabled: true
+      system_probe_config:
+        enabled: false
+        bpf_debug: false
+        debug_port: 0
+        enable_conntrack: false
+        sysprobe_socket: '/opt/datadog-agent/run/sysprobe.sock'
+      EOF
+
+      expect(chef_run).to(render_file('/etc/datadog-agent/system-probe.yaml').with_content { |content|
+        expect(YAML.safe_load(content).to_json).to be_json_eql(YAML.safe_load(expected_yaml).to_json)
+      })
+    end
+  end
+
+  context 'with CWS enabled on Windows' do
+    cached(:solo) do
+      ChefSpec::SoloRunner.new(
+        platform: 'windows',
+        version: '2012R2'
+      ) do |node|
+        node.name 'chef-nodename' # expected to be used as the hostname in `datadog.yaml`
+        node.normal['datadog'] = {
+          'api_key' => 'somethingnotnil',
+          'agent_major_version' => 6,
+          'security_agent' => {
+            'cws' => {
+              'enabled' => true,
+            }
+          }
+        }
+      end
+    end
+
+    cached(:chef_run) do
+      solo.converge(described_recipe) do
+        solo.resource_collection.insert(
+          Chef::Resource::Service.new('datadog-agent', solo.run_context))
+        solo.resource_collection.insert(
+          Chef::Resource::Service.new('datadog-agent-security', solo.run_context))
```

Comment:
If I believe line 76 in `recipes/security-agent.rb` in windows the name changes, correct?
```suggestion
          Chef::Resource::Service.new('datadog-security-agent', solo.run_context))
```

Reply:
I think there is a difference between the "public name" of the service which changes on windows, and the name of the chef service which is common on both platforms